### PR TITLE
WebDataset support

### PR DESCRIPTION
**Pull request by lhoestq**
Added the WebDataset format to the list of supported formats on the Datasets Hub

### CODE REVIEW

File: docs/hub/datasets-image.md [hunk 1]
```diff
@@ -161,4 +182,16 @@ my_dataset_repository/
 └── train.parquet
 ```
 
-Note that for the user convenience, every dataset hosted in the Hub is automatically converted to Parquet format. Read more about it in the [Parquet format](./datasets-viewer#access-the-parquet-files) documentation.
+Images columns are of type struct with a binary field "bytes" for the image data and a string field "path" for the image file name or path.
+Then you can specify the feature types of the columns directly in YAML in the README header, for example:
```
Comment (pcuenca) [inline docs/hub/datasets-image.md:186]:
```suggestion
You can specify the feature types of the columns directly in YAML in the README header, for example:
```

File: docs/hub/datasets-viewer.md [hunk 2]
```diff
@@ -24,17 +24,21 @@ You can search for a word in the dataset by typing it in the search bar at the t
...
-## Access the parquet files
-
-To power the dataset viewer, every dataset is auto-converted to the Parquet format. Click on [_"Auto-converted to Parquet"_](https://huggingface.co/datasets/glue/tree/refs%2Fconvert%2Fparquet/cola) to access the Parquet files. Refer to the [Datasets Server docs](/docs/datasets-server/parquet_process) to learn how to query the dataset parquet files with libraries such as Polars, Pandas or DuckDB.
+## Large scale datasets
 
-You can also access the list of Parquet files programmatically using the [Hub API](./api#endpoints-table): https://huggingface.co/api/datasets/glue/parquet.
+The Dataset Viewer supports large scale datasets, but depending on the data format it may only show the first 5GB of the dataset:
 
-## Very large datasets
+- For Parquet datasets: the Dataset Viewer shows the full dataset, however filtering and search are only enabled on the first 5GB.
```
Comment (pcuenca) [inline docs/hub/datasets-viewer.md:31]:
```suggestion
- For Parquet datasets: the Dataset Viewer shows the full dataset, but filtering and search are only enabled on the first 5GB.
```

File: docs/hub/datasets-viewer.md [hunk 3]
```diff
@@ -24,17 +24,21 @@ You can search for a word in the dataset by typing it in the search bar at the t
...
+- For datasets >5GB in other formats (e.g. [WebDataset](https://github.com/webdataset/webdataset) or JSON Lines): the Dataset Viewer only shows the first 5GB, and filtering and search are enabled on these first 5GB.
 
-For datasets >5GB, we only auto-convert to Parquet the first ~5GB of the dataset. 
 In this case, an informational message lets you know that the Viewer is partial. This should be a large enough sample to represent the full dataset accurately, let us know if you need a bigger sample.
 
+## Access the parquet files
+
+To power the dataset viewer, every dataset is auto-converted up to 5GB to the Parquet format (if the dataset is already in Parquet format we just link all the Parquet files). Click on [_"Auto-converted to Parquet"_](https://huggingface.co/datasets/glue/tree/refs%2Fconvert%2Fparquet/cola) to access the Parquet files. Refer to the [Datasets Server docs](/docs/datasets-server/parquet_process) to learn how to query the dataset parquet files with libraries such as Polars, Pandas or DuckDB.
```
Comment (pcuenca) [inline docs/hub/datasets-viewer.md:38]:
```suggestion
To power the dataset viewer, the first 5G of every dataset are auto-converted to the Parquet format. In the dataset viewer (for example, see [`datasets/glue`](https://huggingface.co/datasets/glue)), you can click on [_"Auto-converted to Parquet"_](https://huggingface.co/datasets/glue/tree/refs%2Fconvert%2Fparquet/cola) to access the Parquet files. Please, refer to the [Datasets Server docs](/docs/datasets-server/parquet_process) to learn how to query the dataset parquet files with libraries such as Polars, Pandas or DuckDB.
```

File: docs/hub/datasets-viewer.md [hunk 4]
```diff
@@ -24,17 +24,21 @@ You can search for a word in the dataset by typing it in the search bar at the t
...
+
+You can also access the list of Parquet files programmatically using the [Hub API](./api#endpoints-table): https://huggingface.co/api/datasets/glue/parquet.
```
Comment (pcuenca) [inline docs/hub/datasets-viewer.md:40]:
```suggestion
You can also access the list of Parquet files programmatically using the [Hub API](./api#get-apidatasetsrepoidparquet); for example, endpoint [`https://huggingface.co/api/datasets/glue/parquet`](https://huggingface.co/api/datasets/glue/parquet) lists the parquet files of the glue dataset.
```

**Review by pcuenca — COMMENTED** [review]
Very cool! 🔥

**Comment by lhoestq** [comment]
Thanks for the suggestions. Cc @severo can you also take a look ? Once this is merged I plan to do some comms on webdataset

**Review by severo — APPROVED** [review]

**Pull request merged**